pkg/nodeconfig/nodeconfig.go

```diff
@@ -255,25 +253,32 @@ func (nc *nodeConfig) configureNetwork() error {
 	return nil
 }
 
-// addVersionAnnotation adds the version annotation to nc.node
-func (nc *nodeConfig) addVersionAnnotation() {
-	nc.node.Annotations[VersionAnnotation] = version.Get()
-}
-
-// addPubKeyHashAnnotation adds the public key annotation to nc.node
-func (nc *nodeConfig) addPubKeyHashAnnotation() {
-	nc.node.Annotations[PubKeyHashAnnotation] = nc.publicKeyHash
-}
-
-// addAdditionalAnnotations merges nc.additionalAnnotations into the annotations on nc.node. If the annotation
-// already existed, its value will be overwritten.
-func (nc *nodeConfig) addAdditionalAnnotations() {
-	if nc.additionalAnnotations == nil {
-		return
+// applyAnnotations applies all the given annotations and updates the Node object in NodeConfig
+// If an annotation already exists, its value will be overwritten.
+func (nc *nodeConfig) applyAnnotations(annotations map[string]string) error {
+	if annotations == nil {
+		return errors.New("expected at least one annotation to apply but received none")
 	}
-	for key, value := range nc.additionalAnnotations {
+	// Accumulate the string representation of a JSON-formatted list of annotation requests
+	// For example: [{"op":"add","path":"/metadata/annotations/foo","value":"bar"},
+	//               {"op":"add","path":"/metadata/annotations/biz","value": "baz"}]
+	patchData := "["
+	for key, value := range annotations {
+		// The `/` in the annotation key needs to be escaped in order to not be considered a "directory" in the path.
+		escapedAnnotation := strings.Replace(key, "/", "~1", -1)
+		patchData += fmt.Sprintf(`{"op":"add","path":"/metadata/annotations/%s","value":"%s"},`,
+			escapedAnnotation, value)
```

Comment:
This logic is being done in multiple places in  our codebase, It might be best to break it off into a separate function in a new package.

Reply:
Annotations pkg created.